File: BackEnd/mimi/src/main/java/com/mimi/Service/PartyService.java
package com.mimi.Service;

import java.util.List;
import java.util.Optional;

import com.mimi.Dto.Party;

public interface PartyService {
	// 모임 업데이트
	public Party save(Party party);
	// 모임 생성
	public Party createParty(Party party);

	// 모임 삭제
	public void deleteParty(Party party);

	// 모임 가져오기
	public Party getParty(String id);
}

File: App/app/src/main/java/com/chd/mimitogether/ui/party/service/PartyService.kt
package com.chd.mimitogether.ui.party.service

import com.chd.mimitogether.ui.party.dto.Party
import com.chd.mimitogether.ui.party.dto.PartyCreate
import retrofit2.Call
import retrofit2.http.*


interface PartyService {

    //모임 생성
    @POST("party/create")
    fun createParty(@Body partyCreate : PartyCreate): Call<Party>

    //유저 id로 party 정보 목록 가져오기
    @GET("party/list/{id}")
    fun getPartyList(@Path("id") id: String): Call<List<Party>>

    @POST("party/join")
    fun joinParty(@Query("partyId") partyId: String,
                  @Query("userId") userId: String): Call<Party>

    @GET("party/{id}")
    fun getPartyDetail(@Path("id") id : String): Call<Party>

}